test-data/unit/stubgen.test

```diff
@@ -4756,3 +4755,122 @@ def polar(*args, **kwargs) -> Incomplete:
 from _typeshed import Incomplete
 
 def polar(*args, **kwargs) -> Incomplete: ...
+
+
+[case testPydanticBaseModel]
+import pydantic
+
+class User(pydantic.BaseModel):
+    id: int
+    name: str
+    active: bool = True
+    optional_field: str | None = None
+[out]
+import pydantic
+
+class User(pydantic.BaseModel):
+    id: int
+    name: str
+    active: bool = ...
+    optional_field: str | None = ...
+
+[case testPydanticBaseModelWithAnnotationsOnly]
+import pydantic
+
+class ConfigSettings(pydantic.BaseModel):
+    # Fields without initialization
+    db_name: str
+    port: int
+    debug: bool
+[out]
+import pydantic
+
+class ConfigSettings(pydantic.BaseModel):
+    db_name: str
+    port: int
+    debug: bool
+
+[case testPydanticNestedBaseModel]
+from pydantic import BaseModel
+
+class Address(BaseModel):
+    street: str
+    city: str
+
+class User(BaseModel):
+    name: str
+    age: int
+    address: Address | None = None
+[out]
+from pydantic import BaseModel
+
+class Address(BaseModel):
+    street: str
+    city: str
+
+class User(BaseModel):
+    name: str
+    age: int
+    address: Address | None = ...
+
+[case testPydanticBaseModelComplex]
+from pydantic import BaseModel
+from typing import Dict, List, Optional, Union
+
+class Item(BaseModel):
+    name: str
+    description: Optional[str] = None
+    tags: List[str] = []
+    properties: Dict[str, Union[str, int, float, bool]] = {}
+[out]
+from pydantic import BaseModel
+
+class Item(BaseModel):
+    name: str
+    description: str | None = ...
+    tags: list[str] = ...
+    properties: dict[str, str | int | float | bool] = ...
+
+[case testPydanticBaseModelInheritance]
+from pydantic import BaseModel
+
+class BaseUser(BaseModel):
+    id: int
+    active: bool = True
+
+class User(BaseUser):
+    name: str
+    email: str
+[out]
+from pydantic import BaseModel
+
+class BaseUser(BaseModel):
+    id: int
+    active: bool = ...
+
+class User(BaseUser):
+    name: str
+    email: str
+
+[case testPydanticModelWithMethods]
```

Comment:
We can just add some methods to existing models. Let's reduce the amount of tests without reducing test features.